numpy/lib/shape_base.py

```diff
@@ -712,7 +759,12 @@ def array_split(ary, indices_or_sections, axis=0):
     return sub_arys
 
 
-def split(ary,indices_or_sections,axis=0):
+def _split_dispatcher(ary, indices_or_sections, axis=None):
```

Comment:
Here you could use a single `_split_dispatcher` with the function above as well as below (where the function currently gets redefined, but then it does get reused a few times).

Reply:
good catch -- fixed